packages/components/src/components/form/Switch/Switch.tsx

```diff
@@ -18,46 +24,39 @@ const Container = styled.div<Pick<SwitchProps, 'isChecked' | 'isDisabled' | 'isS
     transition: background 0.25s ease 0s;
     cursor: ${({ isDisabled }) => !isDisabled && 'pointer'};
     box-sizing: border-box;
+    border: 1px solid ${({ theme, isAlert }) => `${isAlert ? theme.STROKE_RED : 'transparent'}`};
 
-    ${({ isDisabled }) =>
+    button {
+        box-shadow: 0px 2px 4px 0px ${({ theme }) => theme.BOX_SHADOW_BLACK_5};
+        opacity: ${({ isDisabled }) => isDisabled && 0.66};
+    }
+
+    ${({ isDisabled, theme, isChecked }) =>
         !isDisabled &&
         css`
-            :hover,
-            :focus-within {
-                button {
-                    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 1);
-                }
+            :hover {
+                background: ${isChecked ? theme.BG_GREEN_HOVER : theme.STROKE_GREY_HOVER};
             }
-
-            :active {
-                button {
-                    box-shadow: none;
-                }
+            :focus-within {
+                border: 1px solid ${theme.STROKE_BLUE};
+                background: ${isChecked ? theme.BG_GREEN : theme.STROKE_GREY};
+                box-shadow: 0px 0px 0px 3px ${({ theme }) => theme.BOX_SHADOW_SWITCH};
             }
```

Comment:
I believe you can simply use [this](https://github.com/trezor/trezor-suite/blob/9cb5cc576820bd01ced4c5b0501605f3a529fe62/packages/components/src/components/buttons/Button/Button.tsx#L26). Feel free to move it to some other file as you see fit.

Reply:
After changes in the last commit I don't think if I can reuse that much there now

Reply:
Try this, I think it's still nicer to have this unified:

```
export const getFocusShadowStyle = (selector = ':focus-visible') => css`
    border: 1px solid transparent;

    ${selector} {
        border-color: ${({ theme }) => theme.backgroundAlertBlueBold};
        box-shadow: ${boxShadows.focusedLight};
    }

    ${MEDIA_QUERY.DARK_THEME} {
        ${selector} {
            box-shadow: ${boxShadows.focusedDark};
        }
    }
`;
```

And then `getFocusShadowStyle(':focus-within:has(:focus-visible)')`, seems like it's working alright.

Reply:
9ada0fde38e5e788ad4b7eba036e50f07baef91f